src/DDTrace/Integrations/Guzzle/GuzzleIntegration.php

```diff
@@ -138,10 +139,16 @@ private function buildLimitTracerCallback()
      */
     private function setUrlTag(Span $span, $request)
     {
+        $url = null;
```

Comment:
TBH I don't particularly like that a method `setUrlTag` sets both the url tag and overwrite the service name.

What do you think to:

1. Add a `getUrl` that applies the logic to extract the url
2. In `buildPreCallback()` we get the url and if it is bot null we set both the url tag and the service tag?

Reply:
@labbati great suggestion, applied !